### PR TITLE
Cloudinary implementation

### PR DESCRIPTION
**Pull request by granitn**
i hope i didn't tangle up too much.
not sure if i need to add anything to vercel regarding cloudinary

### CODE REVIEW

File: components/MachineCard/index.js [hunk 1]
```diff
@@ -17,13 +18,13 @@ const StyledName = styled.p`
...
-  top: -3rem;
-  box-shadow: var(--shadow-elevation-medium);
-`;
+// const StyledImage = styled(Image)`
+//   object-fit: cover;
+//   border-radius: 50%;
+//   position: relative;
+//   top: -3rem;
+//   box-shadow: var(--shadow-elevation-medium);
+// `;
```
Comment (djfarly) [inline components/MachineCard/index.js]: commented out code does not want to go into main
```suggestion
```

File: components/MachineForm/index.js [hunk 2]
```diff
@@ -33,15 +49,42 @@ export default function MachineForm({ onSubmit, machine }) {
...
+        {machine && (
+          <>
+            <FormLabel htmlFor="image">Picture</FormLabel>
+            <FormInput
+              type="url"
+              name="machineImage"
+              id="image"
+              value={pictureDefault}
+              onChange={(e) => setPictureDefault(e.target.value)}
+              // defaultValue={pictureDefault ? machine?.machineImage : imageUrl}
```
Comment (djfarly) [inline components/MachineForm/index.js]:
```suggestion
```

File: components/MachineForm/index.js [hunk 3]
```diff
@@ -33,15 +49,42 @@ export default function MachineForm({ onSubmit, machine }) {
...
+              placeholder="image url from unsplash"
+              // pattern="(https://images.unsplash.com/.*) | (https://res.cloudinary.com/dko3rskom/.*)"
```
Comment (djfarly) [inline components/MachineForm/index.js]:
```suggestion
```

File: components/MachineForm/index.js [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
+  imageThere,
+  imageUrl,
+}) {
+  const [pictureDefault, setPictureDefault] = useState(
+    imageUrl || machine?.machineImage
+  );
+
+  useEffect(() => {
+    setPictureDefault(imageUrl || machine?.machineImage);
+  }, [imageUrl, machine?.machineImage]);
```
Comment (djfarly) [inline components/MachineForm/index.js:24]: Why is this state called `pictureDefault`? This is the image right? What is default about it?

I don't really get what the effect is supposed to do. Why does the state net to be reset?

File: components/MachineForm/index.js [hunk 5]
```diff
@@ -33,15 +49,42 @@ export default function MachineForm({ onSubmit, machine }) {
...
+            />
+            {machine.machineImage && (
+              <>
+                <label> current image</label>
+                <Image
+                  style={{ objectFit: "cover", borderRadius: "50%" }}
+                  src={machine?.machineImage}
+                  width={100}
+                  height={100}
+                  alt="Cards current image "
```
Comment (djfarly) [inline components/MachineForm/index.js]:
```suggestion
                  alt="Cards current image"
```

File: components/MachineForm/index.js [hunk 6]
```diff
@@ -33,15 +49,42 @@ export default function MachineForm({ onSubmit, machine }) {
...
+          </>
+        )}
+        {imageThere && (
+          <Image
+            style={{ objectFit: "cover", borderRadius: "50%" }}
+            src={imageThere}
+            width={100}
+            height={100}
+            alt="Your uploaded image"
+          ></Image>
```
Comment (djfarly) [inline components/MachineForm/index.js]:
```suggestion
          />
```

File: components/UploadForm/index.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        </p>
+        <ImageInput type="file" id="imageInput" onChange={handleFileChange} />
+        {image && (
+          <Image
+            style={{ objectFit: "cover", borderRadius: "50%" }}
+            src={URL.createObjectURL(image)}
+            width={100}
+            height={100}
+            alt="Image to be uploaded"
+          ></Image>
```
Comment (djfarly) [inline components/UploadForm/index.js:88]:
```suggestion
          />
```

File: components/MachineForm/index.js [hunk 8]
```diff
@@ -1,12 +1,28 @@
...
   StyledForm,
 } from "./MachineForm.Styled";
+import { useState } from "react";
+import Image from "next/image";
+
+export default function MachineForm({
+  onSubmit,
+  machine,
+  imageThere,
+  imageUrl,
```
Comment (djfarly) [inline components/MachineForm/index.js]: What is the reason for `imageThere` and `imageUrl`? What does `imageThere` mean?

**Review by djfarly — CHANGES_REQUESTED** [review]
Hej Granit,

I think I know what you mean by "hacky". I mostly wrote some questions I had when looking at the code.
Please consider make some variable names clearer.

Cheers

File: components/MachineForm/index.js (hunk 4 above)
Comment (granitn) [reply components/MachineForm/index.js:24]: thats a good question indeed i got so tangled up at some point🫣 
and im sure there is a better way to solve it.
i think i used it to make the defaultValue of the type=url input of the MachineForm rendered on the edit page change the displayed url once an image was uploaded.
which got kind of obsolete once i added the image preview but i couldnt let go of it after sacrificing my brain cells for 🥸

File: components/MachineForm/index.js (hunk 8 above)
Comment (granitn) [reply components/MachineForm/index.js]: had to try it out a little myself.
and i came to the conclusion that it must have been a relic from when i only got the add form to work.
took imageThere out of the code.
i must have used it to dynamically render the image preview after uploading

**Comment by granitn** [comment]
hi Jan,
thank you.
I appreciate your feedback and suggestions.
hope what i did was at least somewhat legitimate, while it works as i wanted it to, i'm not to sure about my implementation

next time i should take the time an add comments inline, makes the reviewers work more pleasant and helps myself get a deeper understanding of what i'm coding 😅

**Review by djfarly — APPROVED** [review]
Hej, if some variables/props are relics please consider removing unnecessary code. You have my okay to merge this for now and do a refactor/cleanup later on.

**Pull request merged**